pkg/metadata/scheduler.go

```diff
@@ -20,25 +20,38 @@ import (
 // Catalog keeps track of metadata collectors by name
 var catalog = make(map[string]Collector)
 
+var (
+	// For testing purposes
+	newTimer = time.NewTimer
+	firstRun = true
```

Comment:
you might want to look into using `sync.Once` instead. Also, I don't think we ever set `firstRun` to `false` after calling `scheduler.firstRun()`  (maybe I missed something).

Reply:
It is only set to false in the tests, so that 'first run' code doesn't run when testing. Since this seems confusing, what if I rename it to `runFirstRun` or something less redundant like `enableFirstRunCollection`?